pkgs/build-support/build-bazel-package/default.nix

```diff
@@ -109,13 +113,44 @@ in stdenv.mkDerivation (fBuildAttrs // {
   buildPhase = fBuildAttrs.buildPhase or ''
     runHook preBuild
 
+    '' + lib.optionalString stdenv.isDarwin ''
+    # Bazel sandboxes the execution of the tools it invokes, so even though we are
```

Comment:
This comment is not accurate because Bazel for `buildBazelPackage` is built with Nix hacks. Sorry, I didn't fully get why doesn't this work on Darwin - can you re-explain so that we can put this in a comment?

Reply:
Darwin packages don't always take advantage of the nix environment variables, see  https://github.com/NixOS/nixpkgs/pull/41914 for example.

Reply:
I see, I'll update the comment then as this is already merged.

Reply:
Sorry, I realized I'm still missing something while writing a comment. For example in https://github.com/NixOS/nixpkgs/pull/41914 you set such a variable. If I reason correctly it means that our cc-wrapper works the same for Darwin and can interpret these variables. The code in this PR explicitly converts Nix envvars into additional flags for Bazel. This should only be needed if Nix envvars do not work for Darwin at all, because otherwise they would be passed thanks to Nix hacks.

What am I missing here? Sorry, I don't know anything about Darwin so please bear with me.

Reply:
That PR I did for a separate issue, not related to this change in `buildBazelPackage`. It's just an example of how the wrappers don't forward flags on Darwin. I'm not sure where in the chain of wrappers the forwarding isn't implemented correctly, last time I tried to find all the spots I found a couple of them, then assumed it wasn't supposed to work on Darwin to begin with (a lot of derivations set compiler and linker flags for Darwin explicitly).

So rather than trying to fix the Darwin wrappers to work like the Linux wrappers I implemented this mechanism of specifying the flags explicitly.

Reply:
Ah, I understand the problem now - so the flags are broken somewhere down the line (perhaps in cc-wrapper) and this is a workaround for it. This sounds unfortunate, maybe our Darwin maintainers should take a look?

@NixOS/darwin-maintainers

Reply:
Was the merge okay or should we revert?

Reply:
I think we should wait and see what the Darwin maintainers have to say. Either we revert and fix it for the whole Darwin or we just update a comment (we can do that without a revert).

Reply:
There were a lot of packages that specify flags for Darwin explicitly last time I checked, I don't think we should keep a regression in `buildBazelPackage` around (it used to work before https://github.com/NixOS/nixpkgs/pull/64904) just because of that.

Reply:
I agree; what I mean is that someone has an idea how to just this globally we could go for that instead.